docs/docs/usage.md

```diff
@@ -2,4 +2,93 @@
+restart: always
+networks:
+#? if service.frontend[1].preselected == false {
+#  - frontend-backend
+#? }
+ports:
+  - ${{SPRING_MAVEN_PORT}}:8080
+env_file:
+#? if service.frontend[0] contains name != "spring-maven" {
+#   - environment.env
+#? }
+```
+
+Here is an example for a input for a single statement list: <br>
+`has this.is.a.test.key | this.is contains a.test.key == "value" | this.is.test >= 42`
+
+The cli defaults the mode to the input file, however you can switch to single mode by using the `--mode-single` or `-m` flag for the cli.
+
+## Data path vs data string
+CCom accepts data in the JSON format only, but you can give it to the cli in one of two different forms: <br>
+- Data file: If you have the json input data in form of a file, you can set `--data` / `-d` to the relative or absolute path to the data file. CCom will then load the file from this path.
+- Data string: If you do not have the data in form of a file, you might want CCom to consume a string. For that you can set `--data` / `-d` to a JSON string (e.g.: `{"test-attribute": "test-value"}`)
+
+## Language
+As CCom does not know the content type of your input file, you sometimes have to specify the files language or the comment identifiers. CCom is smart enough to recognize the language for some file extensions, but if the file contents does not match the extension, you can set the language manually via`--lang` / `-l`. This should work out for following languages:
+
+- `yaml`
+- `java`
+- `go`
+- `c`
+- `cpp`
+- `javascript`
+- `typescript`
+- `dockerfile`
+- `html`
+- `xml`
+- `rust`
+- `assembly`
+- `python`
+- more to come ...
+
+If you did not found your language on the list, you can set the comment identifiers by hand. Here is an example: <br>
+`ccom --data ./test-data.json --line-comment-iden "//" --block-comment-iden-open "/*" --block-comment-iden-close "*/" ./test-file.java`
+
+## Switch compilers
+CCom has multiple compiler backend implementations, which have different strengths and weaknesses. The default compiler backend is called `cpp`.
+To switch between compiler implementations, you can use `--compiler` / `-c`. Valid values are `cpp` and `java`.
+
+## Benchmarking
+CCom comes with a built-in benchmark mode to test the performance of compiling different files in combination with different compiler backends. To run a benchmark for a input file with input data, you can use `--benchmark` / `-b` and specify a number of runs. <br>
+E.g.: `ccom  --data ./test-data.json --benchmark 500 --out-file ./output.yml ./input-file.yml`
+
+CCom will show you how long every run takes as well as the total and average runtime.
+
+## Silent mode
+The default behavior of CCom is, to print a few information about compile status to the console. If you're using CCom for your purpose, you might not want such compile status outputs. To disable all CCom outputs, except the compile result output, you can use the `--silent` or `-s` flag for the cli.
```

Comment:
I suggest: `By default, CCom prints some information about the compile status to the console.`